internal/envtest/envtest.go

```diff
@@ -3,69 +3,87 @@ package envtest
 import (
 	"context"
 	"path/filepath"
-	"runtime"
 	"strings"
 	"testing"
 
 	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
 	controlapi "github.com/k8ssandra/cass-operator/apis/control/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
 	k8ssandrataskapi "github.com/k8ssandra/k8ssandra-operator/apis/control/v1alpha1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
 
-	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
-	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/kubectl/pkg/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
 func Run(m *testing.M, setupFunc func(e *Environment)) (code int) {
-	env := NewEnvironment()
-	env.start()
+	ctx := ctrl.SetupSignalHandler()
+	env := NewEnvironment(ctx)
+	env.Start()
 	setupFunc(env)
 	exitCode := m.Run()
-	env.stop()
+	env.Stop()
 	return exitCode
 }
 
 type Environment struct {
-	intClient     client.Client
+	Client        client.Client
 	env           *envtest.Environment
 	cancelManager context.CancelFunc
 	Context       context.Context
+	Kubeconfig    string
```

Comment:
question: Same here, do we need to access it directly?

Reply:
We could write getters and setters, but what benefit does that bring exactly?

Reply:
It's not accessed directly either.